sherwin classes testcases

thepowersss opened the pull request:
not sure if this is the right branch to make a pull request?

thepowersss committed:
classes testcases

justinnhli commented:
Yes it's the right branch. Several of your tests have the same issue: if you declare `class a`, you can't do `a.b` (since `b` is a *instance* variable); you should do `a().b` instead. Fix those and tag me in a comment.

thepowersss committed:
fixed instances vs classes

justinnhli commented:
Mostly good, a few errors:

* `class_in_func.lang` and `integer.lang` have the same class/instance issue. Note that you'll have to fix the sexp as well.
* `class_integer_add.sexp` is (blatantly) wrong.
* The output of class_order_test and singly_linked_list are wrong

thepowersss committed:
fixed minor corrections

- class_in_func and integer .lang and .sexp fixed
- class_integer_add.sexp corrected
- class_order_test.out and single_linked_list.out corrected

thepowersss committed:
Merge remote-tracking branch 'origin/post-deadline' into sherwin-testcases-branch

justinnhli commented:
Classes integer.lang still gets a `member of non-object` error. Be very careful about what `Integer()` returns, and try again?

thepowersss committed:
corrected integer.lang

justinnhli commented:
on the right track, but not there yet. keep trying ;)

thepowersss committed:
integer.lang fix 2

Integer()() since Integer is a function which returns a class?

thepowersss committed:
transforms testcases part 1

thepowersss committed:
transforms testcases part 2

thepowersss committed:
typechecking testcases part 1

justinnhli commented:
Only one minor issue so far: Transforms zero edge case 1.sexp2 is wrong, since even in the sexp there are no negative numbers.

thepowersss committed:
corrected negative sign in zero_edge_case_1.sexp2

thepowersss committed:
typechecking testcases part 2

justinnhli commented:
* Transforms zero edge case 1.sexp2 has a parenthesis error
* Classes interchange var int.sexp should have `varloc` instead of `lookup` for some variables
* Classes typed func call mismatch.out is wrong. `a` *is* a function, with a *return value* that is an int, so the program works fine without type errors.

thepowersss committed:
corrected errors in transforms and typechecking

- added missing paren in zero_edge_case_1.sexp2
- interchange_var_int.sexp replaced lookup with varloc
- corrected func_call_mismatch testcase

justinnhli commented:
Transforms zero edge case 1.sexp2 still has a parenthesis error - double check the scope of each one. That's the only error though!

thepowersss committed:
zero edge case 1 sexp2 corrected parenthesis

justinnhli commented:
merged!
